src/pywifes/splice.py

```diff
@@ -0,0 +1,330 @@
+        self.flux, self.header = fits.getdata(fitsFILE, 0, header=True)
+        self.wl = (
+            np.arange(self.header["NAXIS1"]) - self.header["CRPIX1"] + 1
+        ) * self.header["CDELT1"] + self.header["CRVAL1"]
+        # Temporary
+        self.fluxvar = fits.getdata(fitsFILE, 1, header=False)
+        self.minWL = np.min(self.wl)
+        self.maxWL = np.max(self.wl)
+        return
+
+
+def join_spectra(blueSpec, redSpec):
+    # Generate the resampling matrices
+    blueCRVAL1 = blueSpec.header["CRVAL1"]
+    blueCDELT1 = blueSpec.header["CDELT1"]
+    blueCRPIX1 = blueSpec.header["CRPIX1"]
+    blueNAXIS1 = blueSpec.header["NAXIS1"]
+
+    redCRVAL1 = redSpec.header["CRVAL1"]
+    redCDELT1 = redSpec.header["CDELT1"]
+    redCRPIX1 = redSpec.header["CRPIX1"]
+    redNAXIS1 = redSpec.header["NAXIS1"]
+
+    nwl = int((redCRVAL1 + redCDELT1 * redNAXIS1 - blueCRVAL1) / blueCDELT1)
+    wl = (np.arange(nwl) - blueCRPIX1 + 1) * blueCDELT1 + blueCRVAL1
+
+    # One does not need to interplolate the blue spectra if the waveelength is
+    # set to the blue spectrum. However the code is robust to this.
+
+    AB = _A_lanczos(blueSpec.wl, wl, 3).tocsr()
+    AR = _A_lanczos(redSpec.wl, wl, 3).tocsr()
+
+    flux_B = np.array(AB * blueSpec.flux).ravel()
+    flux_R = np.array(AR * redSpec.flux).ravel()
+
+    # Two sparse diagonal matrices
+    diag_B = sp.dia_matrix(([blueSpec.fluxvar], [0]), shape=[blueNAXIS1, blueNAXIS1])
+    diag_R = sp.dia_matrix(([redSpec.fluxvar], [0]), shape=[redNAXIS1, redNAXIS1])
+
+    fluxvar_B = np.array((AB * diag_B * AB.T).sum(axis=1)).ravel()
+    fluxvar_R = np.array((AR * diag_R * AR.T).sum(axis=1)).ravel()
+
+    buffer = 10.0
+
+    # Should check for wavelngth overlap
+
+    blueOnly = np.where(wl <= redSpec.minWL + buffer)
+    overlap = np.where((wl < blueSpec.maxWL - buffer) & (wl > redSpec.minWL + buffer))
+    redOnly = np.where(wl >= blueSpec.maxWL - buffer)
+
+    if redSpec.minWL > blueSpec.maxWL:
+        return None, None
+
+    else:
+        # Average the two taking into account the buffer region and weighting
+
+        flux = np.zeros(len(flux_B), float)
+        fluxVar = np.zeros(len(flux_B), float)
+
+        flux[blueOnly] = flux_B[blueOnly]
+        fluxVar[blueOnly] = fluxvar_B[blueOnly]
+
+        fluxVar[overlap] = 1.0 / (1.0 / fluxvar_B[overlap] + 1.0 / fluxvar_R[overlap])
+        flux[overlap] = (
+            flux_B[overlap] / fluxvar_B[overlap] + flux_R[overlap] / fluxvar_R[overlap]
+        ) * fluxVar[overlap]
+
+        flux[redOnly] = flux_R[redOnly]
+        fluxVar[redOnly] = fluxvar_R[redOnly]
+
+        return flux, fluxVar
+
+
+def splice_spectra(blue_spec_path, red_spec_path, output):
+    """
+    The main routine
+    """
+    # Create to instances of SingleSpec
+    blueSpec = SingleSpec(blue_spec_path)
+    redSpec = SingleSpec(red_spec_path)
+
+    # Join the spectra
+    print("Splicing spectra")
+    flux, fluxVar = join_spectra(blueSpec, redSpec)
```

Comment:
As I commented on `join_cubes`, careful of it returning None, None. Similar fix to that comment.

Reply:
Yeah, completely right. I updated as  `if flux == None`